File: ruby/quicksort.rb
def quicksort(array)
	pivot = array.pop
	puts pivot
	left = []
	right = []
	return array if array.length < 2
	array.each do |x|
		x <= pivot ? left << x : right << x
	end
	quicksort(left) + [pivot] + quicksort(right)
end

puts quicksort([9,2,4,2,182,23992,1282,4,2,4,6,3,1,2])

File: practice/algos/quick_sort/quick_sort.py
def quick_sort(arr):

    if len(arr) < 2:
        return arr

    pivot = arr.pop()
    left = []
    right = []

    for num in arr:
        if num > pivot:
            right.append(num)
        else:
            left.append(num)
    return quick_sort(left) + [pivot] + quick_sort(right)

arr = [1,2,2,1,2,1,5,23,1,91,2,4,1,]

print(quick_sort(arr))
